ZAPDUtils/Utils/MemoryStream.cpp

```diff
@@ -30,12 +30,19 @@ uint64_t MemoryStream::GetLength()
 
 void MemoryStream::Seek(int32_t offset, SeekOffsetType seekType)
 {
-	if (seekType == SeekOffsetType::Start)
-		baseAddress = offset;
-	else if (seekType == SeekOffsetType::Current)
-		baseAddress += offset;
-	else if (seekType == SeekOffsetType::End)
-		baseAddress = bufferSize - 1 - offset;
+	switch (seekType) {
+		case SeekOffsetType::Start:
+			baseAddress = offset;
+			break;
+
+		case SeekOffsetType::Current:
+			baseAddress += offset;
+			break;
+
+		case SeekOffsetType::End:
+			baseAddress = bufferSize - offset - 1;
+			break;
+	}
```

Comment:
I understand using switch statements as replacements for long if else chains, but it seems there are instances where the switch statements become more bloated than the if-else chains they replace.

Reply:
This one I concede doesn't offer much of an improvement, but since the enum's there anyway, it seemed worth it to me. I don't mind reverting this one.